in method:bad, at line:31, inst:3 = new <Application,Ltestcases/CWE89_SQL_Injection/s01/CWE89_SQL_Injection__database_executeQuery_61b>@0
in method:bad, at line:31, inst:invokespecial < Application, Ltestcases/CWE89_SQL_Injection/s01/CWE89_SQL_Injection__database_executeQuery_61b, <init>()V > 3 @4 exception:4
in method:bad, at line:31, inst:6 = invokevirtual < Application, Ltestcases/CWE89_SQL_Injection/s01/CWE89_SQL_Injection__database_executeQuery_61b, badSource()Ljava/lang/String; > 3 @7 exception:5
in method:bad, at line:33, inst:9 = invokestatic < Application, Ltestcasesupport/IO, getDBConnection()Ljava/sql/Connection; > @18 exception:8
in method:bad, at line:34, inst:11 = invokeinterface < Application, Ljava/sql/Connection, createStatement()Ljava/sql/Statement; > 9 @23 exception:10
in method:bad, at line:35, inst:12 = new <Application,Ljava/lang/StringBuilder>@30
in method:bad, at line:39, inst:invokespecial < Application, Ljava/lang/StringBuilder, <init>()V > 12 @34 exception:13
in method:bad, at line:39, inst:16 = invokevirtual < Application, Ljava/lang/StringBuilder, append(Ljava/lang/String;)Ljava/lang/StringBuilder; > 12,14 @39 exception:15
in method:bad, at line:40, inst:18 = invokevirtual < Application, Ljava/lang/StringBuilder, append(Ljava/lang/String;)Ljava/lang/StringBuilder; > 16,6 @43 exception:17
in method:bad, at line:40, inst:21 = invokevirtual < Application, Ljava/lang/StringBuilder, append(Ljava/lang/String;)Ljava/lang/StringBuilder; > 18,19 @48 exception:20
in method:bad, at line:40, inst:23 = invokevirtual < Application, Ljava/lang/StringBuilder, toString()Ljava/lang/String; > 21 @51 exception:22
in method:bad, at line:40, inst:25 = invokeinterface < Application, Ljava/sql/Statement, executeQuery(Ljava/lang/String;)Ljava/sql/ResultSet; > 11,23 @54 exception:24
in method:badSource, at line:42, inst:6 = invokestatic < Application, Ltestcasesupport/IO, getDBConnection()Ljava/sql/Connection; > @10 exception:5
in method:badSource, at line:47, inst:9 = invokeinterface < Application, Ljava/sql/Connection, prepareStatement(Ljava/lang/String;)Ljava/sql/PreparedStatement; > 6,7 @17 exception:8
in method:badSource, at line:50, inst:11 = invokeinterface < Application, Ljava/sql/PreparedStatement, executeQuery()Ljava/sql/ResultSet; > 9 @24 exception:10
in method:badSource, at line:50, inst:14 = invokeinterface < Application, Ljava/sql/ResultSet, getString(I)Ljava/lang/String; > 11,12 @34 exception:13
in method:badSource, at line:51, inst:conditional branch(eq, to iindex=26) 11,4
in method:badSource, at line:51, inst:invokeinterface < Application, Ljava/sql/ResultSet, close()V > 11 @47 exception:52
in method:badSource, at line:51, inst:54 = getstatic < Application, Ltestcasesupport/IO, logger, <Application,Ljava/util/logging/Logger> >
in method:badSource, at line:51, inst:55 = getstatic < Application, Ljava/util/logging/Level, WARNING, <Application,Ljava/util/logging/Level> >
in method:badSource, at line:54, inst:invokevirtual < Application, Ljava/util/logging/Logger, log(Ljava/util/logging/Level;Ljava/lang/String;Ljava/lang/Throwable;)V > 54,55,31,53 @67 exception:56
in method:badSource, at line:54, inst:conditional branch(eq, to iindex=38) 9,4
in method:badSource, at line:54, inst:invokeinterface < Application, Ljava/sql/PreparedStatement, close()V > 9 @75 exception:58
in method:badSource, at line:67, inst:conditional branch(eq, to iindex=50) 6,4
in method:badSource, at line:67, inst:invokeinterface < Application, Ljava/sql/Connection, close()V > 6 @103 exception:59
in method:badSource, at line:73, inst:61 = getstatic < Application, Ltestcasesupport/IO, logger, <Application,Ljava/util/logging/Logger> >
in method:badSource, at line:73, inst:62 = getstatic < Application, Ljava/util/logging/Level, WARNING, <Application,Ljava/util/logging/Level> >
in method:badSource, at line:70, inst:invokevirtual < Application, Ljava/util/logging/Logger, log(Ljava/util/logging/Level;Ljava/lang/String;Ljava/lang/Throwable;)V > 61,62,38,60 @123 exception:63
in method:badSource, at line:72, inst:19 = getstatic < Application, Ltestcasesupport/IO, logger, <Application,Ljava/util/logging/Logger> >
in method:badSource, at line:72, inst:20 = getstatic < Application, Ljava/util/logging/Level, WARNING, <Application,Ljava/util/logging/Level> >
in method:badSource, at line:72, inst:invokevirtual < Application, Ljava/util/logging/Logger, log(Ljava/util/logging/Level;Ljava/lang/String;Ljava/lang/Throwable;)V > 19,20,21,15 @141 exception:22
in method:badSource, at line:72, inst:conditional branch(eq, to iindex=69) 18,4
in method:badSource, at line:72, inst:invokeinterface < Application, Ljava/sql/ResultSet, close()V > 18 @151 exception:40
in method:badSource, at line:77, inst:42 = getstatic < Application, Ltestcasesupport/IO, logger, <Application,Ljava/util/logging/Logger> >
in method:badSource, at line:77, inst:43 = getstatic < Application, Ljava/util/logging/Level, WARNING, <Application,Ljava/util/logging/Level> >
in method:badSource, at line:79, inst:invokevirtual < Application, Ljava/util/logging/Logger, log(Ljava/util/logging/Level;Ljava/lang/String;Ljava/lang/Throwable;)V > 42,43,31,41 @171 exception:44
in method:badSource, at line:79, inst:conditional branch(eq, to iindex=81) 17,4
in method:badSource, at line:85, inst:invokeinterface < Application, Ljava/sql/PreparedStatement, close()V > 17 @179 exception:46
in method:badSource, at line:84, inst:conditional branch(eq, to iindex=93) 16,4
in method:badSource, at line:84, inst:invokeinterface < Application, Ljava/sql/Connection, close()V > 16 @207 exception:47
in method:badSource, at line:84, inst:49 = getstatic < Application, Ltestcasesupport/IO, logger, <Application,Ljava/util/logging/Logger> >
in method:badSource, at line:84, inst:50 = getstatic < Application, Ljava/util/logging/Level, WARNING, <Application,Ljava/util/logging/Level> >
in method:badSource, at line:89, inst:invokevirtual < Application, Ljava/util/logging/Logger, log(Ljava/util/logging/Level;Ljava/lang/String;Ljava/lang/Throwable;)V > 49,50,38,48 @227 exception:51
in method:badSource, at line:58, inst:return 64
